File: EmojiPanel/src/main/java/com/binlee/emoji/helper/UrlHelper.java
package com.binlee.emoji.helper;

import android.text.TextUtils;

import java.io.File;

public class UrlHelper {

    private UrlHelper() {
        throw new AssertionError("no instance.");
    }

    public static String inspectUrl(String url) {
        if (TextUtils.isEmpty(url)) {
            return url;
        }
        if (isFile(url) || isHttp(url) || isAndroid(url)) {
            return url;
        }
        if (!isHttp(url) && url.startsWith("/")) {
            url = baseUrl() + url;
        } else if (!isHttp(url) && !url.startsWith("/")) {
            url = baseUrl() + File.separator + url;
        }
        return url;
    }

    public static boolean isFile(String url) {
        return url != null && url.startsWith("file://");
    }

    public static boolean isHttp(String url) {
        return url != null && url.startsWith("http");
    }

    public static boolean isAndroid(String url) {
        return url != null && url.startsWith("android.");
    }

    public static String baseUrl() {
        return "";
    }
}

File: EmojiPanel/src/main/java/com/binlee/emoji/helper/UrlHelper.kt
package com.binlee.emoji.helper

import android.text.TextUtils
import java.io.File

class UrlHelper private constructor() {

  companion object {

    fun inspectUrl(url: String?): String? {
      var out = url
      if (TextUtils.isEmpty(out)) {
        return out
      }
      if (isFile(out) || isHttp(out) || isAndroid(out)) {
        return out
      }
      if (!isHttp(out) && out!!.startsWith("/")) {
        out = baseUrl() + out
      } else if (!isHttp(out) && !out!!.startsWith("/")) {
        out = baseUrl() + File.separator + out
      }
      return out
    }

    fun isFile(url: String?): Boolean {
      return url != null && url.startsWith("file://")
    }

    fun isHttp(url: String?): Boolean {
      return url != null && url.startsWith("http")
    }

    fun isAndroid(url: String?): Boolean {
      return url != null && url.startsWith("android.")
    }

    fun baseUrl(): String {
      return ""
    }
  }

  init {
    throw AssertionError("no instance.")
  }
}